artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompFrame.java

```diff
@@ -103,7 +103,7 @@ public ActiveMQBuffer toActiveMQBuffer() throws Exception {
             return buffer;
          }
 
-         StringBuilder head = new StringBuilder();
+         StringBuilder head = new StringBuilder(512);
```

Comment:
Stumbling upon this stuff / code review was how I found the improvement opportunity, not through profiling. I did benchmark the StringBuilder initial size change separately, and it alone provided a bit over half of the total 15% I reported (don't have the exact numbers around any more).